test/browser/fixtures/expect-dom/toMatchScreenshot.test.ts

```diff
@@ -421,4 +421,114 @@ describe('.toMatchScreenshot', () => {
 
     expect(errorMessage).matches(/^Could not capture a stable screenshot within 100ms\.$/m)
   })
+
+  // Only run this test if snapshots aren't being updated
+  test.runIf(server.config.snapshotOptions.updateSnapshot !== 'all')(
+    'runs only once after resolving the element',
+    async ({ onTestFinished }) => {
+      const filename = `toMatchScreenshot-runs-only-once-after-resolving-the-element-1`
+      const path = join(
+        '__screenshots__',
+        'toMatchScreenshot.test.ts',
+      )
+
+      const screenshotPath = join(
+        path,
+        `${filename}-${server.browser}-${server.platform}.png`
+      )
+
+      // Create baseline screenshot with original colors
+      renderTestCase([
+        'oklch(39.6% 0.141 25.723)',
+        'oklch(40.5% 0.101 131.063)',
+        'oklch(37.9% 0.146 265.522)',
+      ])
+      const locator = page.getByTestId(dataTestId)
+
+      await locator.screenshot({
+        save: true,
+        path: screenshotPath,
+      })
+
+      onTestFinished(async () => {
+        await server.commands.removeFile(screenshotPath)
+      })
+
+      // Remove element, then re-render with inverted colors after a delay
+      document.body.innerHTML = ''
+
+      const renderDelay = 500
+      setTimeout(() => {
+        renderTestCase([
+          'oklch(37.9% 0.146 265.522)',
+          'oklch(40.5% 0.101 131.063)',
+          'oklch(39.6% 0.141 25.723)',
+        ])
+      }, renderDelay)
+
+      const start = performance.now()
+
+      // Expected behavior:
+      // 1. `expect.element()` polls until element exists (~500ms)
+      // 2. `toMatchScreenshot()` runs ONCE and fails (colors don't match baseline)
+      //
+      // If `toMatchScreenshot()` polled internally, it would retry for 30s.
+      // By checking the elapsed time we verify it only ran once.
+
+      let errorMessage: string
+
+      try {
+        await expect.element(locator).toMatchScreenshot()
+      } catch (error) {
+        errorMessage = error.message
+      }
+
+      expect(typeof errorMessage === 'string')
```

Comment:
This doesn't do anything, did you mean `expect.assert`? We should maybe throw an error if the methods are not called 😄

Reply:
Editing fail on my part, it was an `if` before but forgot to change `===` with `).toBe(` 😅

Good catch, thanks!